app/react/Library/components/TableRow.tsx

```diff
@@ -0,0 +1,170 @@
+      break;
+    case 'link':
+      result = getLink(prop.value.url, prop.value.label);
+      break;
+    case 'relationship':
+      result = prop.value.map((p: any, index: number) => (
+        <React.Fragment key={index}>
+          {index > 0 && ', '}
+          {getLink(p.url, p.value)}
+        </React.Fragment>
+      ));
+      break;
+    case 'geolocation':
+      result = <GeolocationViewer points={prop.value} onlyForCards />;
+      break;
+    default:
+      result = undefined;
+      break;
+  }
+  return result;
+}
+
+function formatDocument(document: any, templates: TemplateSchema[], thesauris: ThesaurusSchema[]) {
+  let formattedDoc = document.toJS();
+  const template = templates.find((t: TemplateSchema) => t.get('_id') === formattedDoc.template);
+  formattedDoc.templateName = template?.get('name');
+  formattedDoc = formatter.prepareMetadata(formattedDoc, templates, thesauris, null, {
+    sortedProperty: 'creationDate',
+  });
+  formattedDoc.metadata.forEach((prop: any) => {
+    formattedDoc[prop.name] = prop;
+  });
+  return formattedDoc;
+}
+
+function displayCell(
+  document: any,
+  column: any,
+  index: number,
+  firstColumnCheckbox: any,
+  zoomLevel: number
+) {
+  const property = document[column.get('name')];
+  const cellValue = formatProperty(property);
+
+  const otherColumn = (value: any) => (
+    <div className={`table-view-cell table-view-row-zoom-${zoomLevel}`}>{value}</div>
+  );
+  const firstColumn = (value: any) => (
+    <div>
+      {firstColumnCheckbox(index)}
+      {otherColumn(value)}
+    </div>
+  );
+
+  return (
+    <td className={!index ? 'sticky-col' : ''} key={index}>
+      {!index ? firstColumn(cellValue) : otherColumn(cellValue)}
+    </td>
+  );
+}
+
+class TableRowComponent extends Component<TableRowProps> {
+  constructor(props: TableRowProps) {
+    super(props);
+    this.onClick = this.onClick.bind(this);
+    this.firstColumnCheckbox = this.firstColumnCheckbox.bind(this);
+  }
+
+  onClick(e: { preventDefault: () => void }) {
+    if (this.props.onClick) {
+      this.props.onClick(e, this.props.document, this.props.selected);
+    }
+  }
+
+  firstColumnCheckbox(index: number) {
+    return (
+      !index && (
+        <input type="checkbox" checked={this.props.selected} onClick={this.onClick.bind(this)} />
+      )
+    );
+  }
+
+  render() {
+    const formattedDocument = formatDocument(
+      this.props.document,
+      this.props.templates,
+      this.props.thesauris
+    );
+
+    return (
+      <tr className={this.props.selected ? 'selected' : ''}>
+        {this.props.columns.map((column: any, index: number) =>
+          displayCell(
+            formattedDocument,
+            column,
+            index,
+            this.firstColumnCheckbox,
+            this.props.zoomLevel
+          )
+        )}
+      </tr>
+    );
+  }
+}
+
+function mapStateToProps(state: any, ownProps: TableRowProps) {
+  const selected = ownProps.storeKey
```

Comment:
storeKey is defined as either 'library' or 'uploads' in the Props definition.  Yet here, there is a tertiary contemplating the option of not having storeKey?  If this is a possibility, please include it as conditional in the props definition.  If not, then why this tertiary?